modules/atomic-widgets/style/style-render.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace Elementor\Modules\AtomicWidgets\Style;
+
+class Style_Render {
+	private array $styles;
+	private array $transformers;
+	private array $breakpoints;
+
+	public function __construct( array $styles, array $transformers, array $breakpoints ) {
+		$this->styles = $styles;
+		$this->transformers = $transformers;
+		$this->breakpoints = $breakpoints;
+	}
+
+	public function render(): string {
+		$css_style = [];
+
+		foreach ( $this->styles as $style_def ) {
+			$style = $this->render_style( $style_def );
+			$css_style[] = $style;
+		}
+
+		return implode( '', $css_style );
+	}
+
+	private function render_style( array $style ): string {
+		$base_selector = $this->get_base_selector( $style );
+
+		if ( ! $base_selector ) {
+			return '';
+		}
+
+		$stylesheet = [];
+
+		foreach ( $style['variants'] as $variant ) {
+			$style_declaration = $this->variant_to_style_declaration( $base_selector, $variant );
+
+			if ( $style_declaration ) {
+				$stylesheet[] = $style_declaration;
+			}
+		}
+
+		return implode( '', $stylesheet );
+	}
+
+	private function get_base_selector( array $style_def ): string {
+		$map = [
+			'class' => '.',
+		];
+
+		return $map[ $style_def['type'] ] . $style_def['id'];
+	}
```

Comment:
what about cases where the `type` is not in the map for some reason?
let's add some validations, and maybe remove the map for now until we actually need it

Reply:
As the TS code also has a similar flow, so for the sake of consistency  - I'll keep the map for now
But added validations 🙂